ironfish/src/rpc/adapters/httpAdapter.ts

```diff
@@ -0,0 +1,145 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import http from 'http'
+import { Assert } from '../../assert'
+import { Logger } from '../../logger'
+import { ErrorUtils } from '../../utils'
+import { RpcRequest } from '../request'
+import { ApiNamespace, Router } from '../routes'
+import { RpcServer } from '../server'
+import { IRpcAdapter } from './adapter'
+
+export class RpcHttpAdapter implements IRpcAdapter {
+  server: http.Server | null = null
+  router: Router | null = null
+
+  readonly host: string
+  readonly port: number
+  readonly logger: Logger
+  readonly namespaces: ApiNamespace[]
+  readonly maxRequestSize: number
+
+  // TODO(daniel): implement https + basic authentication with rpcToken
```

Comment:
we can delete this, but after it merges we could enable HTTPS using the certs that exist